packages/x-components/src/x-modules/experience-controls/store/actions/fetch-and-save-controls.action.ts

```diff
@@ -0,0 +1,29 @@
+import { createFetchAndSaveActions } from '../../../../store/utils/fetch-and-save-action.utils';
+import { ExperienceControlsActionContext } from '../types';
+
+const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
+  ExperienceControlsActionContext,
+  any,
+  any
+>({
+  fetch({ dispatch }, request) {
+    return dispatch('fetchControls', request);
```

Comment:
When you have the full adapter here you won't just receive the controls but also the events to emit. This should be called `fetchExperienceControlsResponse`